PKPD/gui/simulation.py

```diff
@@ -680,31 +764,58 @@ def fill_parameter_slider_group(self):
-        self.slider_min_max_label_container = [] # store in list to be able to update later
-        self.parameter_text_field_container = [] # store in list to be able to update later
-        for param_id, param_name in enumerate(parameter_names):
-            self.parameter_sliders.addWidget(self._create_slider(param_name), param_id, 0)
+        # length of parameters so can fill up in correct order later
+        self.slider_container = [None]*len(parameter_names)  # store in list to be able to update later
+        self.slider_min_max_label_container = [None]*len(parameter_names) # store in list to be able to update later
+        self.parameter_text_field_container = [None]*len(parameter_names) # store in list to be able to update later
+
+        # fill up grid with inferred parameter boxes
+        self.inferred_boxes = [None] * len(parameter_names)
+
+        # Get unique list of components and sort alphabetically for consistent display
+        collapse_boxes = list(set([p.split('.')[0] for p in parameter_names]))
+        collapse_boxes.sort()
+
+        # Create box for each component:
+        # TODO Group parameters earlier to make this more efficient
+        for name in collapse_boxes:
+            box = CollapsibleBox("{}".format(name.replace('_', ' ').capitalize()))
+            self.parameter_sliders.addWidget(box) #add to parameter slider section
+            lay = QtWidgets.QGridLayout()
+            # Add correct parameters to each box component
+            # But keeping original ID labelling so as not to affect inference etc.
+            for param_id, param_name in enumerate(parameter_names):
+                if param_name.split('.')[0] == name:
+                    lay.addWidget(self._create_slider(param_name, param_id), param_id, 0)
+            box.setContentLayout(lay)
+            #self.parameter_sliders.addStretch()
+            self.parameter_sliders.setAlignment(QtCore.Qt.AlignTop)  # display nicely
+
+
 
         # initialise container to store parameter values (for efficiency)
         number_parameters = len(self.parameter_text_field_container)
         self.parameter_values = np.empty(number_parameters)
 
 
     def _clear_slider_group(self):
-        """Clears the slider group from pre-existing sliders.
         """
-        number_items_in_group = self.parameter_sliders.count()
-        for item_id in range(number_items_in_group):
-            # setting an items parent to None deletes it, according to stackoverflow
-            self.parameter_sliders.itemAtPosition(item_id, 0).widget().setParent(None)
+        Clears the slider group from pre-existing sliders.
+        """
+        # Solution from Stackoverflow
+        # - using takeAt & deleteLater vs. itemAt & setParent(None) which didn't work in this case
+        for _ in range(self.parameter_sliders.count()):
+            child = self.parameter_sliders.takeAt(0)
+            if child.widget():
```

Comment:
If possible, could you remove the if child.widget or, alternatively, use a try/except block. This would prevent problems with silent failures if something unexpected gets passed to this function.